rfcs/0001-shabka-usage-design.md

```diff
@@ -0,0 +1,186 @@
+
+The `shabka` CLI will be used by the user to do what he can currently do using
+the tools is the `scripts/` directory. This will allow him to do everything he
+could do using the `nixos-rebuild` command, plus building a host, checking what
+changed in his host's generation between two git refs, and maybe even pushing
+his hosts' configurations to Cachix.
+
+Here is what the CLI usage might look like:
+
+### `shabka rebuild`
+
+This will do what `scripts/nixos-rebuild.sh` currently does. Its usage is:
+
+```sh
+shabka rebuild [-h host] [-r release] {nixos-rebuild arguments and options}
+```
+
+By default the `host` argument is the current hostname, the `release` argument
+is the host's release (defined in `.shabka/hosts/{host}/.release`) if
+defined, or `shabka`'s default, defined in `shabka/.release`.
+
+This command takes care of setting all environment variables and options to
+ensure a smooth `nixos-rebuild` depending on what is defined in `shabka`,
+`.shabka` and `dotshabka-user`. The default locations of `.shabka` and
+`dotshabka-user` are defined in their respective section of this document.
+
+### `shabka build`
+
+This will do what `scripts/build-host.sh` currently does. Its usage is:
+
+```sh
+shabka build [-r release] [host]
+```
+
+By default the `host` argument is the current hostname, the `release` argument
+is the host's release (defined in `.hosts/hosts/{host}/.release`) if
+defined, or `shabka`'s default, defined in `shabka/.release`.
+
+> This command cannot build a host with a different kernel than the host it is
+> being ran on, e.g. a `Darwin` host cannot be built on a `Linux` host.
+
+This command takes care of setting all environment variables and options to
+ensure a smooth `nix-build` depending on what is defined in `shabka`, `.shabka`
+and `dotshabka-user`. The default locations of `.shabka` and `dotshabka-user`
+are defined in their respective section of this document.
+
+### `shabka diff`
+
+This will do what `scripts/diff-host.sh` used to do. Its usage is:
+
+```sh
+shabka diff <host> <base-rev> [target-rev]
+```
+
+By default the `target-rev` argument is HEAD.
+
+/!\ This section needs improvement given the fact that we now have 2
+repositories to deal with (`shabka` and `.shabka`, as `dotshabka-user` is
+defined in `.shabka`). This implies a lot of arguments and some designing
+around this command must be done.
+
+### `shabka push-to-cachix`
+
+This will do what `scripts/push-to-cachix.sh` currently does. Its usage is:
+
+```sh
+shabka push-to-cachix <cache-name> [host1] [host2] [host3] [...]
+```
+
+If no host is given, this command will push every host in `.shabka/hosts`. This
+command requires the environment variable `CACHIX_SIGNING_KEY` to be set.
+
+## `dotshabka-user` for a user configuration
+
+This directory will be used to define a user's personal configuration.
+
+This directory must have a `home.nix` that defines a
+function for the user as its now done in the `home.nix` inside a host's
```

Comment:
The current host configuration in `dotshabka` does include a `home.nix` file, but this file is not actually imported by shabka, instead directly by the Host's configuration: https://github.com/kalbasit/dotshabka/blob/faed8dacea0b5b441ee1a24d71a2324022b63a7a/hosts/hades/configuration.nix#L23

I can inline `home.nix` in the `configuration.nix` and shabka would still build:

```
  imports = [
    ./hardware-configuration.nix

    "${shabka.external.nixos-hardware.path}/common/cpu/intel"
    "${shabka.external.nixos-hardware.path}/common/pc/laptop"
    "${shabka.external.nixos-hardware.path}/common/pc/laptop/ssd"

    <shabka/modules/nixos>

    ({
        shabka.home-manager.config = { userName, uid, isAdmin, home, nixosConfig }:
        { ... }:

        {
            shabka.workstation.enable = true;
        }
    })

    ../../modules/nixos
  ]
```

Reply:
This is discussed in an other thread.